crates/fuel-streams-publisher/src/transactions.rs

```diff
@@ -0,0 +1,52 @@
+use fuel_core::combined_database::CombinedDatabase;
+use fuel_core_storage::transactional::AtomicView;
+use fuel_streams_core::{
+    prelude::IntoSubject,
+    transactions::TransactionsSubject,
+    types::{
+        BlockHeight,
+        ChainId,
+        Transaction,
+        TransactionKind,
+        TransactionStatus,
+        UniqueIdentifier,
+    },
+    Stream,
+};
+use tracing::info;
+
+pub async fn publish(
+    chain_id: &ChainId,
+    block_height: &BlockHeight,
+    fuel_core_database: &CombinedDatabase,
+    transactions_stream: &Stream<Transaction>,
+    transactions: &[Transaction],
+) -> anyhow::Result<()> {
+    let off_chain_database = fuel_core_database.off_chain().latest_view()?;
+
+    for (transaction_index, transaction) in transactions.iter().enumerate() {
+        let tx_id = transaction.id(chain_id);
+        let kind = TransactionKind::from(transaction.to_owned());
+        let status: TransactionStatus = off_chain_database
+            .get_tx_status(&tx_id)?
+            .map(|status| status.into())
+            .unwrap_or_default();
+
+        let transactions_subject: TransactionsSubject =
```

Comment:
Cant we do From trait here to just use .into() ?

Reply:
No, it won't work that way since the source type is an `Option<T>`

Reply:
Not sure what you mean, isnt this just a mapping of T -> S. Event with options you couldo do: .map(Into::into)

Reply:
Oh, sorry. Missed the line you were referring to. My earlier comment was about `TransactionStatus`. 

For this subject, we need external data to compose it correctly, namely, `chain_id`, `block_height`, et al, and they can't be gotten from `Transaction`. That said, we do need a better abstraction for these transformations, but it is one of the things I decided to defer. 👍🏾

Reply:
OK, fair enough with me!